src/lib/hyrise.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include "boost/container/pmr/memory_resource.hpp"
+#include "concurrency/transaction_manager.hpp"
+#include "storage/storage_manager.hpp"
+#include "utils/plugin_manager.hpp"
+#include "utils/singleton.hpp"
+
+namespace opossum {
+
+class Hyrise : public Singleton<Hyrise> {
+ public:
+  // Resets the Hyrise state by deleting the Plugin-, Storage-, and TransactionManager
+  // and creating new ones. This is used especially in tests and can lead to a lot of
+  // issues if there are still running tasks / threads that want to access a resource.
+  // You should be very sure that this is what you want. Have a look at base_test.hpp
+  // to see the correct order of resetting things.
```

Comment:
The "correct order" thing should be a thing of the past now, shouldn't it?

Reply:
I thought that "correct order" meant setting the CurrentScheduler to nullptr first to be sure that all tasks are done and then resetting Hyrise. But the sentence might still be misleading, so I'd be fine with just dropping it.

Reply:
Ah, makes sense. Never mind then.